File: LMS_CMS_Backend/LMS_CMS_DAL/Models/Domains/HR/MonthlyAttendance.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LMS_CMS_DAL.Models.Domains.HR
{
    public class MonthlyAttendance : AuditableEntity
    {
        public long Id { get; set; }

        public DateOnly Day { get; set; }
        public double WorkingHours { get; set; } 
        public int LateHours { get; set; }
        public int LateMinutes { get; set; }
        public int OvertimeHours { get; set; }
        public int OvertimeMinutes { get; set; }
        public int LeaveRequestHours { get; set; }
        public int LeaveRequestMinutes { get; set; }

        public long EmployeeId { get; set; }
        public Employee Employee { get; set; }

        public long DayStatusId { get; set; }
        public DayStatus DayStatus { get; set; }


    }
}

File: LMS_CMS_Frontend/src/app/Models/HR/monthly-attendance.ts
export class MonthlyAttendance {
    constructor(
        public id: number = 0,
        public day: string = '',

        public workingHours: number = 0,
        public workingMinutes: number = 0,
        public deductionHours: number = 0,
        public deductionMinutes: number = 0,
        public overtimeHours: number = 0,
        public overtimeMinutes: number = 0,
        public leaveRequestHours: number = 0,
        public leaveRequestMinutes: number = 0,

        public employeeId: number = 0,
        public employeeEnName: string = '',
        public employeeArName: string = '',

        public dayStatusId: number = 0,
        public dayStatusName: string = '',

        public insertedAt: string = "",
        public insertedByUserId: number = 0,
    ) { }
}

